### PR TITLE
Restricted featuregate reader

### PR DESCRIPTION
**Pull request by deads2k**
Update library-go to get a more restrictive featuregate reader to avoid naughty usage.

### CODE REVIEW

File: pkg/operator/configobservation/cloud/observe_cloud_volume_plugin.go [hunk 1]
```diff
@@ -10,10 +11,20 @@ import (
 	"github.com/openshift/library-go/pkg/operator/events"
 )
 
+func NewOObserveCloudVolumePluginFunc(featureGateAccessor featuregates.FeatureGateAccess) configobserver.ObserveConfigFunc {
```
Comment (benluddy) [inline pkg/operator/configobservation/cloud/observe_cloud_volume_plugin.go:14]:
```suggestion
func NewObserveCloudVolumePluginFunc(featureGateAccessor featuregates.FeatureGateAccess) configobserver.ObserveConfigFunc {
```

**Comment by deads2k** [comment]
external ccm bug

**Comment by benluddy** [comment]
/lgtm
/hold

Holding for a workloads team review.

**Comment by deads2k** [comment]
/hold cancel

if there is a problem, we can revert, though library-go will eventually need udpating.

**Pull request merged**